parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -440,7 +470,15 @@ public static final ParquetMetadata readFooter(Configuration configuration, File
     }
   }
 
-  private static final ParquetMetadata readFooter(FileStatus file, FSDataInputStream f, MetadataFilter filter) throws IOException {
+  /**
+   * Reads the meta data block in the footer of the file using provided input stream
+   * @param f input stream for the file
+   * @param file the parquet File
+   * @param filter the filter to apply to row groups
+   * @return the metadata blocks in the footer
+   * @throws IOException if an error occurs while reading the file
+   */
+  public static final ParquetMetadata readFooter(FileStatus file, FSDataInputStream f, MetadataFilter filter) throws IOException {
```

Comment:
We only need need the file size here not the full FileStatus object.
(since we're making this public)